enforcer.go

```diff
@@ -437,7 +437,11 @@ func (e *Enforcer) enforce(matcher string, explains *[]string, rvals ...interfac
 				for _, ruleName := range ruleNames {
 					if j, ok := parameters.pTokens[ruleName]; ok {
 						rule := util.EscapeAssertion(pvals[j])
-						expWithRule = util.ReplaceEval(expWithRule, rule)
+						if strings.Contains(rule, ">") || strings.Contains(rule, "<") || strings.Contains(rule, "=") {
```

Comment:
Why change this? This seems to restrict the evaluated rule to be boolean, and if it's not boolean, then return "false". But I originally thought that the evaluated rule actually can return any type like boolean or int, or string. Of course the user should take care of the types, to match the return type with its occurrence in the matcher.

Reply:
Once user use RBAC with ABAC model, we have to make sure `eval` slot is boolean.

Reply:
OK. Let's do this first. See customer effect, maybe this is not an issue.